content-services/content-sources-backend
# Fixes 1738: Unify on context for dao & pulp layer

## Summary
Unifies context passing into methods across Dao and PulpClient layers from handler and tasks.  Also sets up the logger to properly pull the request ID from the context. Now the context is passed fully from:
* the handler, to the dao, to the pulp_client
* the handler to the task to the doa & pulp client

All dao db calls should now have a .WithContext() call to add the context to the db call.  

## Testing steps

Run a curl request to create (and snapshot a repo):

```
UUID="1cf4b0b0-fc0a-11ee-8913-201e8814f721"
curl -H "x-rh-insights-request-id:$UUID"  http://localhost:8000/api/content-sources/v1/repositories/  -H "`./scripts/header.sh 1 1`"   -d '{"name":"foo", "url":"https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/", "snapshot":true}'  -H "Content-Type: application/json"
```

And watch the server logs:

you should see a lot of:
``` request_id=1cf4b0b0-fc0a-11ee-8913-201e8814f721```
Both at the api level and then as the task runs

Also check  the pulp workers logs:

```podman logs cs_pulp_worker_1 ```
and you should see the same request id there:
```
pulp [1cf4b0b0-fc0a-11ee-8913-201e8814f721]: pulpcore.tasking.tasks:INFO: Task completed 018ee838-baf4-7997-aa91-a4f24bcd82d9
```

## Checklist
